link/src/test/java/com/stripe/android/link/ui/signup/SignUpViewModelTest.kt

```diff
@@ -0,0 +1,437 @@
+package com.stripe.android.link.ui.signup
+
+import androidx.navigation.NavHostController
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.Logger
+import com.stripe.android.core.model.CountryCode
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.link.LinkActivityContract
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.link.LinkScreen
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.analytics.LinkEventsReporter
+import com.stripe.android.link.model.LinkAccount
+import com.stripe.android.link.model.StripeIntentFixtures
+import com.stripe.android.link.ui.inline.SignUpConsentAction
+import com.stripe.android.model.ConsumerSession
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.doSuspendableAnswer
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.stub
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.RobolectricTestRunner
+import kotlin.time.Duration.Companion.milliseconds
+
+@RunWith(RobolectricTestRunner::class)
+internal class SignUpViewModelTest {
+    private val dispatcher = UnconfinedTestDispatcher()
+    private val config = LinkConfiguration(
+        stripeIntent = StripeIntentFixtures.PI_SUCCEEDED,
+        merchantName = MERCHANT_NAME,
+        merchantCountryCode = "",
+        customerInfo = LinkConfiguration.CustomerInfo(
+            name = CUSTOMER_NAME,
+            email = CUSTOMER_EMAIL,
+            phone = CUSTOMER_PHONE,
+            billingCountryCode = CUSTOMER_BILLING_COUNTRY_CODE
+        ),
+        shippingValues = null,
+        flags = emptyMap(),
+        cardBrandChoice = null,
+        passthroughModeEnabled = false
+    )
+    private val defaultArgs = LinkActivityContract.Args(
+        configuration = config,
+    )
+
+    private val navController: NavHostController = mock()
+    private val linkAccountManager: LinkAccountManager = mock()
+    private val linkEventsReporter: LinkEventsReporter = mock()
+    private val logger = mock<Logger>()
```

Comment:
can you use fakes or real implementations for these instead of mocks, unless there is some reason that isn't possible?

Reply:
LinkAccountManager and NavController aren't interfaces so we can't have fakes for this. I'm using fakes for logger and `linkEventsReporter` now.

Reply:
I suspect it's a fairly simple refactor to make LinkAccountManager an interface and it's likely worth it, given how many tests are modifying its behavior and validating that it's called correctly. 

You can refactor like this:
- rename the existing LinkAccountManager to something like RealLinkAccountManager or DefaultLinkAccountManager
- add an interface called LinkAccountManager, which RealLinkAccountManager implements
- use a fake in the new tests you added